pkg/pf/tests/provider_configure_test.go

```diff
@@ -30,27 +30,77 @@ import (
 )
 
 func TestConfigure(t *testing.T) {
-    t.Parallel()
-	t.Run("string", crosstests.MakeConfigure(
+	t.Parallel()
+
+	t.Run("string", testConfigurePrimitive{
+		zeroValue:    cty.StringVal(""),
+		nonZeroValue: cty.StringVal("a string"),
+		attrOptional: schema.StringAttribute{Optional: true},
+		attrRequired: schema.StringAttribute{Required: true},
+	}.run)
+
+	t.Run("bool", testConfigurePrimitive{
+		zeroValue:    cty.BoolVal(false),
+		nonZeroValue: cty.BoolVal(true),
+		attrOptional: schema.BoolAttribute{Optional: true},
+		attrRequired: schema.BoolAttribute{Required: true},
+	}.run)
+
+	t.Run("int", testConfigurePrimitive{
+		zeroValue:    cty.NumberIntVal(0),
+		nonZeroValue: cty.NumberIntVal(123),
+		attrOptional: schema.Int64Attribute{Optional: true},
+		attrRequired: schema.Int64Attribute{Required: true},
+	}.run)
+
+	t.Run("float", testConfigurePrimitive{
+		zeroValue:    cty.NumberFloatVal(0),
+		nonZeroValue: cty.NumberFloatVal(123.5),
+		attrOptional: schema.Float64Attribute{Optional: true},
+		attrRequired: schema.Float64Attribute{Required: true},
+	}.run)
+
+	t.Run("secret-string", crosstests.MakeConfigure(
 		schema.Schema{Attributes: map[string]schema.Attribute{
 			"k": schema.StringAttribute{Optional: true},
 		}},
 		map[string]cty.Value{"k": cty.StringVal("foo")},
+		crosstests.ConfigurePulumiConfig(resource.PropertyMap{"k": resource.MakeSecret(resource.NewProperty("foo"))}),
 	))
+}
 
-	t.Run("secret-string", crosstests.MakeConfigure(
+type testConfigurePrimitive struct {
+	zeroValue, nonZeroValue    cty.Value
+	attrOptional, attrRequired schema.Attribute
+}
+
+func (tc testConfigurePrimitive) run(t *testing.T) {
+	t.Parallel()
```

Comment:
I think this needs t.Helper() to make the stack trace for the failure point to the actual parameters